pkg/karmadactl/cmdinit/config/config.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2024 The Karmada Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	"os"
+	"sort"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	yamlserializer "k8s.io/apimachinery/pkg/runtime/serializer/yaml"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/klog/v2"
+)
+
+// LoadInitConfiguration loads the InitConfiguration from the specified file path.
+// It delegates the actual loading to the loadInitConfigurationFromFile function.
+func LoadInitConfiguration(cfgPath string) (*KarmadaInitConfig, error) {
+	var config *KarmadaInitConfig
+	var err error
+
+	config, err = loadInitConfigurationFromFile(cfgPath)
+
+	return config, err
+}
+
+// loadInitConfigurationFromFile reads the file at the specified path and converts it into an InitConfiguration.
+// It reads the file contents and then converts the bytes to an InitConfiguration.
+func loadInitConfigurationFromFile(cfgPath string) (*KarmadaInitConfig, error) {
+	klog.V(1).Infof("loading configuration from %q", cfgPath)
+
+	b, err := os.ReadFile(cfgPath)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read config from %q: %v", cfgPath, err)
+	}
+
+	return BytesToInitConfiguration(b)
+}
+
+// BytesToInitConfiguration parses the given byte slice into an InitConfiguration.
+// It first converts the bytes to a map of GroupVersionKind to bytes, then processes this map to extract the InitConfiguration.
+func BytesToInitConfiguration(b []byte) (*KarmadaInitConfig, error) {
```

Comment:
is it possible to use viper to load config file, like

```
workDir, _ := os.Getwd()
	viper.SetConfigName("config")
	viper.SetConfigType("yml")
	viper.AddConfigPath(path.Join(workDir, "./"))
	//viper.AddConfigPath(path.Join(workDir, "config/application.yml"))
	err := viper.ReadInConfig()
```

Reply:
Using viper for parsing is indeed a better choice, and I will change it soon

Reply:
Here, using Viper might not be the best choice. I am still researching how to better use the Kubernetes API library to handle different versions of configuration files. It seems that Viper may not be particularly suitable for this scenario. I would appreciate more of your advice.

Reply:
I think it's a better choice to use k8s.io/apimachinery/pkg/runtime/serializer/yaml for parsing here. What do you think? I have tried using Viper for parsing, but the results were not very satisfactory. Using the k8s.io/apimachinery/pkg/runtime/serializer/yaml package for parsing might also facilitate future modifications to support parsing methods for multiple versions of configuration files. What do you think about keeping it as is without making changes here?

Reply:
I think it's a better choice to use `k8s.io/apimachinery/pkg/runtime/serializer/yaml` for parsing here. What do you think? I have tried using Viper for parsing, but the results were not very satisfactory. Using the `k8s.io/apimachinery/pkg/runtime/serializer/yaml` package for parsing might also facilitate future modifications to support parsing methods for multiple versions of configuration files. What do you think about keeping it as is without making changes here?
@liangyuanpeng

Reply:
It will be helpful to link the kubeadm parsing configuration file code here

Reply:
In kubeadm, the parsing of configuration files is done by maintaining a global CodecFactory, which registers different version-specific configuration file decoders within its runtime.Decoder. This approach allows for the parsing of configuration files corresponding to different versions, as shown in the code linked here
https://github.com/kubernetes/kubernetes/blob/4f796c02f77fb95d42cd161ea663dd1bf05e372f/cmd/kubeadm/app/util/config/initconfiguration.go#L338C1-L347C4
https://github.com/kubernetes/kubernetes/blob/4f796c02f77fb95d42cd161ea663dd1bf05e372f/cmd/kubeadm/app/apis/kubeadm/v1beta3/register.go#L31C1-L64C2

As for how the configuration file is translated into the information used in the structures for actual deployment operations, I will address that in a comment below regarding that specific information.